README.md

```diff
@@ -126,40 +136,97 @@ including [views/admin-index.php](src/wordpress/admin/views/admin-index.php).
+reading the WordPress [developer documentation](https://developer.wordpress.org/themes/functionality/localization/)
 and the [i18n for WordPress Developers](https://codex.wordpress.org/I18n_for_WordPress_Developers)
 before following the documentation here: https://wordpress.stackexchange.com/a/258562.
 
-Eventually you will run a command similar to:
+Eventually you will run a command similar to this one to create a pot file:
 
 ```
 cd wpdev/tools/i18n
 php makepot.php wp-plugin ../../../react-wordpress-plugin/src/wordpress/ plugin-name.pot
 ```
 
-Our preferred editor is PoEdit (availble for both Mac and Windows: https://poedit.net/download).
+From there you can start translating with our preferred editor [PoEdit](https://poedit.net/download)
+(availble for both Mac and Windows) for instance.
+
+### i18n Translation in React
+
+To enable translations in React we use [react-intl](https://github.com/yahoo/react-intl).
+
+#### How to use react-intl
+
+Simply use the following components and methods from react-intl:
+[defineMessages()](https://github.com/yahoo/react-intl/wiki/API#definemessages),
+[<FormattedMessage\>](https://github.com/yahoo/react-intl/wiki/Components#formattedmessage), and
+[<FormattedHTMLMessage\>](https://github.com/yahoo/react-intl/wiki/Components#formattedhtmlmessage)
+in your components. Check out the example in the [App](src/components/app/index.js)
+component to get started.
+
+#### How to extract and generate json files for each language
+
+Once you have added some translation strings, you can generate the necessary files
+to start translating. To create the translation files run either `yarn watch` or
+`yarn client`. [babel-plugin-react-intl](https://github.com/yahoo/babel-plugin-react-intl)
+will then create the `*.json` files (by extracting the default message descriptors)
+for each language and our [i18n webpack plugin](webpack/plugins/i18n.js) will do
+the rest (eg. copy it into `dist/static/languages`).
+
+#### How to translate and use translation
+
+All you have to do now is translating the generated \*.json files (see
+`/dist/static/languages/*.json`) and putting them back into the static folder
+before releasing your plugin.
+
+**Attention:** babel-plugin-react-intl will overwrite the translation files in
+`dist/static/languages` everytime it runs. This should be improved in the future,
+as [react-intl-translations-manager](https://www.npmjs.com/package/react-intl-translations-manager)
+is capable of doing this!
+
+#### Additonal scripts and help
+
+To generate only one default language (currently en) file called `en.json` run
+`yarn i18n:generate-default-language` _after_ you built the app.
+
+Because `yarn build` uses the custom made webpack plugin to create multiple
+translation files for multiple languages (see [webpack/common.config.js](webpack/common.config.js)).
+With the command above you can overwrite it.
+
+If you need - for whatever reason - to generate a \*.json file from a \*.po
```

Comment:
is this a "standard workflow" or could this be part of a FAQ/Troubleshooting?

Reply:
@lumannnn not sure, as it is only relevant in the "Additonal scripts and help" section, not in general for the project/app. I'll leave it here for now.